### PR TITLE
update log level setting, close #44

### LINKED ISSUE #44: Log middleware has problem receiving log level setting
When I set it to `:error`
image

When I set it to `:info`
image

But there are all requests with 200 HTTP status code.

This is the log setting in my client:
```elixir
middleware Maxwell.Middleware.Logger, [log_level: :info]
```

### PR DESCRIPTION
**Pull request by falood**

### CODE REVIEW

**Review by zhongwencool — APPROVED** [review]

File: lib/maxwell/middleware/logger.ex [hunk 1]
```diff
@@ -16,47 +16,105 @@ defmodule Maxwell.Middleware.Logger do
...
+      {_, [_|_]} ->
+        raise ArgumentError, "Logger Middleware Options doesn't accept wrong_option (:log_level)"
+      {nil, _}                           -> [default: :info]
+      {options, _} when is_list(options) -> parse_opts(options)
+      {level, _}                         -> parse_opts([{level, :default}])
+    end
   end
 
-  def call(request_env, next_fn, level) do
+  def call(request_env, next_fn, options) do
```
Comment (secretworry) [inline lib/maxwell/middleware/logger.ex:38]: We should move message formatting logic into a separate method, so we can implment the call function more concisely.
```elixir
start_time = :erlang.system_time :micro_seconds
next_fn.(request_env) |> log_request(start_time, options)
```

File: lib/maxwell/middleware/logger.ex [hunk 2]
```diff
@@ -16,47 +16,105 @@ defmodule Maxwell.Middleware.Logger do
...
     start_time = :os.timestamp()
     new_result = next_fn.(request_env)
     method = request_env.method |> to_string |> String.upcase
     case new_result do
       {:error, reason, _conn} ->
         error_reason = to_string(:io_lib.format("~p", [reason]))
```
Comment (secretworry) [inline lib/maxwell/middleware/logger.ex]: I just wonder, why not to use `inspect` here.

File: lib/maxwell/middleware/logger.ex [hunk 3]
```diff
@@ -16,47 +16,105 @@ defmodule Maxwell.Middleware.Logger do
...
+    color =
+      case level do
+        nil    -> nil
+        :debug -> IO.ANSI.cyan
+        :info  -> IO.ANSI.normal
+        :warn  -> IO.ANSI.yellow
+        :error -> IO.ANSI.red
       end
+
+    unless is_nil(color) do
```
Comment (secretworry) [inline lib/maxwell/middleware/logger.ex]: I know `is_nil(color)` is equal to `is_nil(level)`, but the later one would be more straitforward.

File: lib/maxwell/middleware/logger.ex (hunk 1 above)
Comment (falood) [reply lib/maxwell/middleware/logger.ex:38]: I think `call` function is concisely enough, if you still think there're too many logic into it, we can make another pr to fix it.
that's how plug do time format https://github.com/elixir-lang/plug/blob/master/lib/plug/logger.ex#L45-L46

**Review by secretworry — APPROVED** [review]

**Comment by zhongwencool** [comment]
Thanks for the PR!!  👏 🎊
Tomorrow, I will published 2.2.0  with this version   :)

**Pull request merged**